massa-db-worker/src/massa_db.rs

```diff
@@ -1614,4 +1607,119 @@ mod test {
+        db_opts.set_paranoid_checks(true);
+
+        let _db = MassaDB::new_with_options(db_config, db_opts.clone()).unwrap();
+        let db = Arc::new(RwLock::new(
+            Box::new(_db) as Box<(dyn MassaDBController + 'static)>
+        ));
+
+        // Add data 1 (at slot 1)
+        let batch_key_1 = vec![1, 2, 3];
+        let batch_value_1 = vec![4, 5, 6];
+        let batch_1 = DBBatch::from([(batch_key_1.clone(), Some(batch_value_1.clone()))]);
+        let versioning_batch_1 =
+            DBBatch::from([(vec![10, 20, 30], Some(vec![127, 128, 254, 255]))]);
+        let mut guard = db.write();
+        guard.write_batch(batch_1, versioning_batch_1, Some(slot_1));
+        drop(guard);
+
+        // Add data 2 (at slot 1)
+        let batch_key_2 = vec![11, 22, 33];
+        let batch_value_2 = vec![44, 55, 66];
+        let batch_2 = DBBatch::from([(batch_key_2.clone(), Some(batch_value_2.clone()))]);
+        let versioning_batch_2 = DBBatch::from([(vec![12, 23, 34], Some(vec![255, 254, 253]))]);
+        let mut guard = db.write();
+        guard.write_batch(batch_2, versioning_batch_2, Some(slot_1));
+        drop(guard);
+
+        // Stream using StreamingStep::Started
+        let last_state_step: StreamingStep<Vec<u8>> = StreamingStep::Started;
+        let stream_batch_ = db.read().get_batch_to_stream(&last_state_step, None);
+        let stream_batch = stream_batch_.unwrap();
+        assert_eq!(
+            stream_batch.new_elements,
+            BTreeMap::from([
+                (batch_key_1.clone(), batch_value_1.clone()),
+                (batch_key_2.clone(), batch_value_2)
+            ])
+        );
+        assert_eq!(stream_batch.updates_on_previous_elements, BTreeMap::new());
+        assert_eq!(stream_batch.change_id, slot_1);
+
+        // Now updates some values for each slot until slot 2 (included)
+        let mut cur_slot = slot_1.get_next_slot(THREAD_COUNT).unwrap();
+        while cur_slot <= slot_2 {
+            let batch_key = vec![(cur_slot.period % 256) as u8];
+            let batch_value = vec![cur_slot.thread];
+            let batch = DBBatch::from([(batch_key.clone(), Some(batch_value.clone()))]);
+            let versioning_batch = DBBatch::from([(
+                vec![(cur_slot.period % 256) as u8],
+                Some(vec![cur_slot.thread]),
+            )]);
+            let mut guard = db.write();
+            guard.write_batch(batch, versioning_batch, Some(cur_slot));
+            drop(guard);
+            cur_slot = cur_slot.get_next_slot(THREAD_COUNT).unwrap();
+        }
+
+        // Stream using StreamingStep::Finished
+        let last_state_step: StreamingStep<Vec<u8>> =
+            StreamingStep::Finished(Some(batch_key_2.clone()));
+        let stream_batch_ = db
+            .read()
+            .get_batch_to_stream(&last_state_step, Some(slot_1));
+        assert!(stream_batch_.is_ok());
+
+        // Now updates some values for each slot until slot 2 (included)
```

Comment:
Maybe explain why redoing this, will make the cache miss some data?

Reply:
Done